application/frontend/controllers/AuthController.php

```diff
@@ -0,0 +1,123 @@
+<?php
+namespace frontend\controllers;
+
+use Sil\IdpPw\Common\Auth\RedirectException;
+use Sil\IdpPw\Common\Auth\User as AuthUser;
+use common\models\User;
+use yii\helpers\Url;
+use yii\web\Controller;
+use yii\web\UnauthorizedHttpException;
+
+class AuthController extends Controller
+{
+    /**
+     * @param \yii\base\Action $action
+     * @return bool
+     * @throws \yii\web\BadRequestHttpException
+     */
+    public function beforeAction($action)
+    {
+        /*
+         * Disable CSRF validation for login since user is redirected to an IdP for logging in
+         */
+        if ($action->id == 'login') {
+            // @todo can this be changed to use a URL parameter for the
+            // token for this action and we can pass to idp and back?
+            $this->enableCsrfValidation = false;
+        }
+        return parent::beforeAction($action);
+    }
+
+    public function actionLogin()
+    {
+        /*
+         * Collect return to url of where to send user after successful login
+         * Expected as relative url starting with /
+         * Before redirecting user after login this will be prefixed with ui_url
+         */
+        $returnTo = \Yii::$app->request->get('ReturnTo', '');
+
+        if ( ! \Yii::$app->user->isGuest) {
+            $afterLogin = $this->getAfterLoginUrl($returnTo);
+            return $this->redirect($afterLogin);
+        }
+
+        /*
+         * Initialize $log variable for logging
+         */
+        $log = ['action' => 'login'];
+
+        try {
+            $returnTo = Url::to(['auth/login', 'ReturnTo' => $returnTo], true);
+            /** @var AuthUser $authUser */
+            $authUser = \Yii::$app->auth->login(\Yii::$app->request, $returnTo);
+
+            $log['email'] = $authUser->email;
+            // Get local user instance or create one
+            $user = User::findOrCreate(null, null, $authUser->employeeId);
```

Comment:
We're intentionally only using the `employeeId` when finding the user's record when they log in here, right? I wonder if it might be worth adding a comment to remind us of that fact, so that we don't later replace these `null` values with attributes of the `$authUser`.